Runtime/Behaviours/CinemachineStoryboard.cs

```diff
@@ -120,11 +132,24 @@ class CanvasInfo
             if (vcam != VirtualCamera || stage != CinemachineCore.Stage.Finalize)
                 return;
 
+            UpdateRenderCanvas();
+
             if (m_ShowImage)
                 state.AddCustomBlendable(new CameraState.CustomBlendable(this, 1));
             if (m_MuteCamera)
                 state.BlendHint |= CameraState.BlendHintValue.NoTransform | CameraState.BlendHintValue.NoLens;
         }
+        
+        RenderMode m_PreviousRenderMode = RenderMode.ScreenSpaceOverlay;
+        void UpdateRenderCanvas()
+        {
+            if (m_PreviousRenderMode != m_RenderMode)
+            {
+                m_PreviousRenderMode = m_RenderMode;
+                DestroyCanvas();
+            }
+        }
```

Comment:
I could also do:
```
Canvas m_Canvas;
void UpdateRenderCanvas()
{
    if (m_Canvas != null && m_Canvas.renderMode != m_RenderMode)
    {
        m_Canvas.renderMode = m_RenderMode;
        m_Canvas.worldCamera = m_RenderCamera;
    }
}
```

I think it is better to do a fast check, more expensive change, then a slow check and faster change. Because I don't expect people changing their canvas render mode often (probably never in play).